include/string.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 Merzlov Nikolay merzlovnik@mail.ru
+#include "string.hpp"
+
+String::~String() {
+    delete[] Data;
+}
+
+String::String() : size(0) {
+    Data = new char[size];
+}
+
+String::String(const String &rhs) : size(rhs.size) {
+    Data = new char[size];
+    for (unsigned int i = 0; i < size; ++i) {
+        Data[i] = rhs[i];
+    }
+}
+
+String::String(const char *data) : size(0) {
```

Comment:
Instead of complicated magic, simply call other constructor as `String(strlen(data))` and then use `std::copy(..)` to copy text